pandas/core/arrays/string_.py

```diff
@@ -318,16 +327,25 @@ def __init__(self, values, copy=False):
 
     def _validate(self):
         """Validate that we only store NA or strings."""
-        if len(self._ndarray) and not lib.is_string_array(self._ndarray, skipna=True):
-            raise ValueError("StringArray requires a sequence of strings or pandas.NA")
         if self._ndarray.dtype != "object":
             raise ValueError(
                 "StringArray requires a sequence of strings or pandas.NA. Got "
                 f"'{self._ndarray.dtype}' dtype instead."
             )
+        try:
+            lib.ensure_string_array(
+                self._ndarray.ravel("K"),
```

Comment:
I'm pretty sure this is broken for 2D EAs even with the ravel.
@jbrockmendel Do you know how I can create a 2D StringArray to test this? Would the correct way to iterate over a 2D EA be using the ``PyArray_GETITEM`` and ``PyArray_ITER_DATA`` combo like the ``Validator`` in libs does? Then I would replace values by ``PyArray_SETITEM``ing on the array?
(Not familiar at all with Numpy C API, so going to need some help here)

Reply:
> Do you know how I can create a 2D StringArray to test this?

`pd.array(["foo", "bar"] * 5).reshape(5, 2)`

> Would the correct way to iterate over a 2D EA be using the PyArray_GETITEM and PyArray_ITER_DATA combo like the Validator in libs does?

Not sure I understand the question.  The 2D EA has a working `__iter__` method, but you shouldn't be passing any EA to any of the cython functions.

> Then I would replace values by `PyArray_SETITEM` ing on the array?

I'm still getting the hang of this (intending to make array_to_timedelta64 handle 2D directly to avoid ravels), would advise NOT trying to handle it in the cython code.